### PR TITLE
Adding a couple BasicChangeSignature Integration Tests

### PR DESCRIPTION
**Pull request by dpoeschl**
Introduces the ability to verify dialog presence and click buttons in
dialogs.

There are more tests coming. Doing just this much now is to partially unblock @CyrusNajmabadi for his integration tests.

### CODE REVIEW

**Comment by dpoeschl** [comment]
Tagging @dotnet/roslyn-ide

**Comment by dpoeschl** [comment]
Investigating failures. It says that `View.ShowSmartTag` is unavailable, something that worked on my machine and for which the infrastructure existed before this chance. Maybe it only works on fixes and not refactorings?

File: src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    End Sub
+End Class");
+
+            OpenDialog();
+            VerifyChangeSignatureDialog(isOpen: true);
+            ChangeSignatureDialog_ClickCancel();
+            VerifyChangeSignatureDialog(isOpen: false);
+        }
+
+        private void OpenDialog()
```
Comment (tmeschter) [inline src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs]: Maybe rename to `InvokeChangeSignature`?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 2]
```diff
@@ -361,5 +362,93 @@ private static FixAllSuggestedAction GetFixAllSuggestedAction(IEnumerable<Sugges
...
             => ExecuteOnActiveView(view =>  System.Windows.MessageBox.Show(message));
+
+        public void VerifyDialog(string dialogAutomationId, bool isOpen)
+        {
+            var dialogAutomationElement = FindDialog(dialogAutomationId, isOpen);
+
+            if ((isOpen && dialogAutomationElement == null) ||
+                (!isOpen && dialogAutomationElement != null))
+            {
+                throw new InvalidOperationException($"Expected the {dialogAutomationId} dialog to be {(isOpen ? "open" : "closed")}, but it is not.");
```
Comment (tmeschter) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: What about turning these checks and throws into xUnit assertions?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 3]
```diff
@@ -361,5 +362,93 @@ private static FixAllSuggestedAction GetFixAllSuggestedAction(IEnumerable<Sugges
 
         public void MessageBox(string message)
             => ExecuteOnActiveView(view =>  System.Windows.MessageBox.Show(message));
+
+        public void VerifyDialog(string dialogAutomationId, bool isOpen)
+        {
```
Comment (tmeschter) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: Do we need to guarantee that this happens on the UI thread, or is it OK for it to happen on a background thread?

**Comment by dpoeschl** [comment]
@rchande No, it is not a complete replacement. I don't plan to mark that test as completely migrated.

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs (hunk 2 above)
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: I can do that.

File: src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs (hunk 1 above)
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs]: Sure.

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs (hunk 3 above)
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: All automationelement work can be done from a background thread. The only reason this even needs to be in-proc is to get the VS hwnd from DTE.

Comment (tmeschter) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: Good to know.

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs (hunk 2 above)
Comment (tmeschter) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: Err, actually, I'm not sure about that. These are on the VS side, so I'm not sure it really makes sense there.

**Comment by tmeschter** [comment]
@dpoeschl Is there a technical reason that we can't completely replace the XML-based test? Or is that just coming in a future change?

**Comment by dpoeschl** [comment]
@tmeschter Coming in a future change. Working on it now.

File: src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            SetUpEditor(@"
+Class C
+    Sub Method$$(a As Integer, b As String)
+    End Sub
+End Class");
+
+            OpenDialog();
+            VerifyChangeSignatureDialog(isOpen: true);
+            ChangeSignatureDialog_ClickCancel();
+            VerifyChangeSignatureDialog(isOpen: false);
```
Comment (Pilchie) [inline src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs]: Potentially, verify that text didn't change?

Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs]: Sure.

**Pull request merged**